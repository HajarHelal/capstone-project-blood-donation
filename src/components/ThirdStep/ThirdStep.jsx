import React, { useState } from 'react';
import { PropTypes } from 'prop-types';
import { useTranslation } from 'react-i18next';
import Button from '../Button';
import HeaderPharagrph from '../Paragraph';
import SubHeader from '../SubHeader';
import Donateparagraph from '../Donateparagraph';
import Progress from '../Progress';
import InputField from '../InputField';
import donate from './images/donate.svg';

function ThirdStep({ donor, onChange, setStep, onRegister }) {
  const { t } = useTranslation();
  const [agree, setAgree] = useState(false);
  const onClickNext = () => {
    if (donor.name !== '' && donor.email !== '' && donor.phone !== '' && agree) {
      onRegister();
    }
  };
  const onClickPrevious = () => {
    setStep(2);
  };
  return (
    <div>
      <div className=" w-11/12 xl:w-10/12 mb-6 sm:mb-8 md:mb-12">
        <HeaderPharagrph
          headerText={t('donate_third_page.title')}
          headerFlag={false}
          paragraphText={t('donate_third_page.paragrph')}
          paragraphFlag={false}
        />
      </div>
      <div className="p-6 sm:p-8 md:p-10 border-gray-300 shadow-xl border-2 rounded">
        <Progress steps={3} />
        <SubHeader text={t('donate_third_page.subHeader1')} />
        <Donateparagraph text={t('donate_third_page.subparagrph')} />
        <div className="flex items-start ">
          <div className="w-9/12 md:flex-1 ">
            <SubHeader text={t('donate_first_page.subHeader2')} />
            <div className="mb-8 md:mb-16">
              <InputField
                labelText={t('donate_third_page.name')}
                placeholder={t('donate_third_page.name_placeholder')}
                onChange={onChange}
                inputValue={donor.name}
                inputName="name"
                Inputtype="text"
              />
            </div>
            <div className="mb-8 md:mb-16">
              <InputField
                labelText={t('donate_third_page.phone')}
                placeholder={t('donate_third_page.phone_placeholder')}
                onChange={onChange}
                inputValue={donor.phone}
                inputName="phone"
                Inputtype="number"
              />
            </div>
            <div className="mb-8  md:mb-16">
              <InputField
                labelText={t('donate_third_page.email')}
                placeholder={t('donate_third_page.email_placeholder')}
                onChange={onChange}
                inputValue={donor.email}
                inputName="email"
                Inputtype="email"
              />
            </div>
            <div>
              <label htmlFor="check" className="inline-flex items-center">
                <input
                  type="checkbox"
                  className="form-checkbox h-4 w-4 "
                  name="agree"
                  checked={agree}
                  onChange={(e) => setAgree(e.target.checked)}
                />
                <span className="mx-2 text-sm sm:text-base md:text-lg">
                  {t('donate_third_page.agree')}
                </span>
              </label>
            </div>
          </div>
          <div className="hidden md:block md:flex-1 shadow-sm">
            <img
              className="w-10/12 mx-auto object-contain"
              style={{ height: '600px' }}
              src={donate}
              alt="donate"
            />
          </div>
        </div>

        <div className="flex  mb-8">
          <div className="mx-4 sm:mx-8 text-center">
            <Button
              text={t('donate_third_page.button_previous')}
              onclick={onClickPrevious}
              classFlag
            />
          </div>
          <div>
            <Button
              text={t('donate_third_page.button_next')}
              onclick={onClickNext}
              classFlag={false}
            />
          </div>
        </div>
      </div>
    </div>
  );
}
ThirdStep.propTypes = {
  donor: PropTypes.object.isRequired,
  setStep: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  onRegister: PropTypes.func.isRequired,
};
export default ThirdStep;
