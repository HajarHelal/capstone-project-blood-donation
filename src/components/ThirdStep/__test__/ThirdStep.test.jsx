import React from 'react';
import renderer from 'react-test-renderer';
import ThirdStep from '../ThirdStep';

// This is an example of a Snapshot. It will create a new snapshot which all future
// tests will use to compare to.
it('renders correctly', () => {
  const tree = renderer
    .create(
      <ThirdStep
        donor={{
          name: '',
          email: '',
          phone: '',
          date: '',
          weight: '',
          sickness: '',
          bloodType: '',
          gender: '',
        }}
        onChange={() => console.log('donate')}
        setStep={() => console.log('donate')}
        onRegister={() => console.log('donate')}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
