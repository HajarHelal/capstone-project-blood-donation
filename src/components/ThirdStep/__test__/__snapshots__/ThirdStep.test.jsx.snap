// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className=" w-11/12 xl:w-10/12 mb-6 sm:mb-8 md:mb-12"
  >
    <div>
      <h1
        className="text-primary-100  sm:text-xl md:text-2xl lg:text-3xl pt-1 font-semibold  "
      >
        donate_third_page.title
      </h1>
      <p
        className=" pt-3 pb-5  sm:text-sm md:text-base lg:text-lg   "
      >
        donate_third_page.paragrph
      </p>
    </div>
  </div>
  <div
    className="p-6 sm:p-8 md:p-10 border-gray-300 shadow-xl border-2 rounded"
  >
    <div
      className="flex  items-center w-10/12  mb-10"
    >
      <div
        className="text-base md:text-lg lg:text-xl font-semibold  mx-4 sm:mx-8"
      >
        donate_first_page.step
        3 / 3
      </div>
      <div
        className="flex-1 h-6 sm:h-8 md:h-10 bg-gray-100"
      >
        <div
          className="w-4/12 h-6 sm:h-8 md:h-10 hidden bg-primary"
        />
        <div
          className="w-8/12 h-6 sm:h-8 md:h-10 hidden bg-primary"
        />
        <div
          className="w-100 h-6 sm:h-8 md:h-10  bg-primary"
        />
      </div>
    </div>
    <div
      className="text-base md:text-lg lg:text-xl font-medium mb-6"
    >
      donate_third_page.subHeader1
    </div>
    <div
      className="text-sm sm:text-base md:text-lg  mb-10 leading-relaxed w-10/12"
    >
      donate_third_page.subparagrph
    </div>
    <div
      className="flex items-start "
    >
      <div
        className="w-9/12 md:flex-1 "
      >
        <div
          className="text-base md:text-lg lg:text-xl font-medium mb-6"
        >
          donate_first_page.subHeader2
        </div>
        <div
          className="mb-8 md:mb-16"
        >
          <div
            className="flex-1 flex flex-col  mb-4 "
          >
            <label
              className="flex-1 mb-2 text-sm sm:text-base font-semibold"
              htmlFor="name"
            >
              donate_third_page.name
            </label>
            <input
              className="flex-1  mr-4 p-2 text-sm sm:text-base tracking-normal font-semibold  border-2 border-gray-300 focus:outline-none  bg-white placeholder-gray-800"
              id="name"
              name="name"
              onChange={[Function]}
              placeholder="donate_third_page.name_placeholder"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="mb-8 md:mb-16"
        >
          <div
            className="flex-1 flex flex-col  mb-4 "
          >
            <label
              className="flex-1 mb-2 text-sm sm:text-base font-semibold"
              htmlFor="phone"
            >
              donate_third_page.phone
            </label>
            <input
              className="flex-1  mr-4 p-2 text-sm sm:text-base tracking-normal font-semibold  border-2 border-gray-300 focus:outline-none  bg-white placeholder-gray-800"
              id="phone"
              name="phone"
              onChange={[Function]}
              placeholder="donate_third_page.phone_placeholder"
              type="number"
              value=""
            />
          </div>
        </div>
        <div
          className="mb-8  md:mb-16"
        >
          <div
            className="flex-1 flex flex-col  mb-4 "
          >
            <label
              className="flex-1 mb-2 text-sm sm:text-base font-semibold"
              htmlFor="email"
            >
              donate_third_page.email
            </label>
            <input
              className="flex-1  mr-4 p-2 text-sm sm:text-base tracking-normal font-semibold  border-2 border-gray-300 focus:outline-none  bg-white placeholder-gray-800"
              id="email"
              name="email"
              onChange={[Function]}
              placeholder="donate_third_page.email_placeholder"
              type="email"
              value=""
            />
          </div>
        </div>
        <div>
          <label
            className="inline-flex items-start"
            htmlFor="check"
          >
            <input
              checked={false}
              className="form-checkbox h-6 w-6 "
              name="agree"
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="mx-2 text-sm sm:text-base "
            >
              donate_third_page.agree
               
              <a
                className="border-b-2  border-primary text-sm sm:text-base  text-primary"
                href="www.facebook.com"
                target="_blank"
              >
                donate_third_page.terms
              </a>
            </span>
          </label>
        </div>
      </div>
      <div
        className="hidden md:block md:flex-1 shadow-sm"
      >
        <img
          alt="donate"
          className="w-10/12 mx-auto object-contain"
          src="donate.svg"
          style={
            Object {
              "height": "600px",
            }
          }
        />
      </div>
    </div>
    <div
      className="flex  mb-8"
    >
      <div
        className="mx-4 sm:mx-8 text-center"
      >
        <button
          className="inline-block text-primary-100 text-base md:text-lg  text-center md:font-medium bg-white  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95"
          onClick={[Function]}
          type="button"
        >
          donate_third_page.button_previous
        </button>
      </div>
      <div>
        <button
          className="inline-block text-white text-base md:text-lg  text-center md:font-medium bg-primary  md:mt-8 mt-4 px-8 md:px-12 lg:px-16 py-1 lg:py-2  shadow-md transition duration-500 ease-in-out transform hover:scale-95"
          onClick={[Function]}
          type="button"
        >
          donate_third_page.button_next
        </button>
      </div>
    </div>
  </div>
</div>
`;
